sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -1185,6 +1185,32 @@ def create_pipeline(self):
   def test_register_finalizations(self):
     raise unittest.SkipTest("TODO: Avoid bundle finalizations on repeat.")
 
+class FnApiRunnerTestWithMultiWorkers(FnApiRunnerTest):
+
+  def create_pipeline(self):
+    return beam.Pipeline(
+        runner=fn_api_runner.FnApiRunner(num_workers=2))
+
+  def test_checkpoint(self):
+    raise unittest.SkipTest("Multiworker doesn't support split request.")
+
+  def test_split_half(self):
+    raise unittest.SkipTest("Multiworker doesn't support split request.")
+
+class FnApiRunnerTestWithMultiWorkersAndBundleRepeat(FnApiRunnerTest):
+
+  def create_pipeline(self):
+    return beam.Pipeline(
+        runner=fn_api_runner.FnApiRunner(num_workers=2, bundle_repeat=2))
+
+  def test_checkpoint(self):
+    raise unittest.SkipTest("Multiworker doesn't support split request.")
```

Comment:
ParallelBundleManager works well with all tests except the ones with split manager. Without fully knowing about how split request works, I'm not sure what issue it is. I would like to confirm if we want to support pipelines with split request in this case, because splitting is another way to chunk data into smaller set. In addition, depends on how split manager is designed, in some cases, split function will not be called. For example, `test_split_half` test only run split when `num_elements == total_num_elements`, however, with ParalleBundleManager, num_elements will not be equal to total_num_elements because it is already chunked to N pieces.

Reply:
Here is a pattern I observed with `test_checkpoint` test. If we split the buffer to N pieces, there are (N-1) elements repeat twice.
For example, if we split [a,b,c] into 2 pieces, it returns one of [a,a,b,c] or [a,b,b,c] or [a,b,c,c]. If we split [a,b,c] into 3 pieces, it returns one of [a,a,b,b,c] or [a,a,b,c,c] or [a,b,b,c,c]. if we split [a,b,c,d] into 2 pieces, it returns one of [a,a,b,c,d] or [a,b,b,c,d] or [a,b,c,c,d] or [a,b,c,d,d].

Reply:
This isn't an issue anymore with a new class with partition() function.